ChildProject/pipelines/metrics.py

```diff
@@ -20,26 +20,65 @@ def __init__(
         recordings: Union[str, List[str], pd.DataFrame] = None,
         from_time: str = None,
         to_time: str = None,
+        rec_cols: str = None,
+        child_cols: str = None,
     ):
 
         self.project = project
         self.am = ChildProject.annotations.AnnotationManager(self.project)
 
-        recording_columns = {
+        #necessary columns to construct the metrics
+        join_columns = {
             "recording_filename",
             "child_id",
             "duration",
             "session_id",
             "session_offset",
         }
-        recording_columns &= set(self.project.recordings.columns)
-
+        #get existing columns of the dataset for recordings
+        correct_cols = set(self.project.recordings.columns)
+        if by not in correct_cols: exit("<{}> is not specified in this dataset, cannot extract from it".format(by))
```

Comment:
can you please add some text to the user, so that they have an idea of what they can try to do? If I read this correctly, we should say "required column not specified, please check the spelling" or something like that?

Reply:
lets change it to : <{}> is not specified in this dataset, cannot extract by it, change your --by option